gen-json-schema-class.py

```diff
@@ -0,0 +1,225 @@
+        "--inline", "--no-metadata",
+        "--title-from=title",
+        "--closed", str(tpl)
+    ])
+    if not schema_str:
+        print(f"⚠️  Skipping {cls}")
+        continue
+
+    try:
+        raw = json.loads(schema_str)
+    except json.JSONDecodeError as e:
+        print(f"❌ Invalid JSON for {cls}: {e}")
+        continue
+
+    # set ID + title
+    raw["$id"]   = (
+        f"https://repo-prod.prod.sagebase.org/repo/v1/schema/type/registered/"
+        f"sagedm-{cls.lower()}"
+    )
+    raw["title"] = cls
+
+    # --- dereference, but leave $refs intact so we can inline enums ourselves
+    deref = jsonref.replace_refs(raw, merge_props=False, proxies=False)
+    full  = deref
+
+    # pull out all enum definitions
+    defs  = full.pop("$defs", {})
+    props = full.setdefault("properties", {})
+
+    # walk every property; when we hit a $ref to an enum, inline only its enum list + title
+    def resolve(obj):
+        if isinstance(obj, dict):
+            if "$ref" in obj:
+                ref       = obj.pop("$ref")
+                enum_name = ref.rsplit("/", 1)[-1]
+                enumdef   = defs.get(enum_name)
+                if enumdef and "enum" in enumdef:
+                    # merge only the enum values
+                    obj["enum"] = enumdef["enum"]
+                    # bring over human-friendly title if you set one
+                    if "title" in enumdef:
+                        obj["title"] = enumdef["title"]
+                return
+            for v in obj.values():
+                resolve(v)
+        elif isinstance(obj, list):
+            for item in obj:
+                resolve(item)
+
+    resolve(props)
+
+    # now normalize, order, and write out as before
+    def normalize(o):
+        if isinstance(o, dict):
+            t = o.get("type")
+            if isinstance(t, list) and set(t) == {"string","null"}:
+                o["type"] = "string"
+            for v in o.values():
+                normalize(v)
+        elif isinstance(o, list):
+            for i in o:
+                normalize(i)
+
+    normalize(full)
+
+    ordered = OrderedDict()
+    for k in ["$id","$schema","title","type","description","properties","required"]:
+        if k in full:
+            ordered[k] = full[k]
+    for k,v in full.items():
+        if k not in ordered:
+            ordered[k] = v
+
+    remove_keys(ordered, {"additionalProperties","metamodel_version","version"})
+
+    out_json = OUT_DIR / f"{cls}-deref.json"
+    out_json.write_text(json.dumps(ordered, indent=2))
+    print(f"  ✅ wrote {out_json}")
+
+
+# ─── STEP 4: VALIDATE AGAINST SYNAPSE (dryRun) ───────────────────────────────
+print("\n🔨 Validating all dereferenced schemas…")
+syn = synapseclient.Synapse()
+syn.login()
+
+def validate_schema(path: Path):
+    print(f"\n🔍 Validating: {path.name}")
+    data = json.loads(path.read_text())
+    remove_keys(data, {"additionalProperties","metamodel_version","version"})
```

Comment:
This is great but shouldn't have `remove_keys` here since that was already done in the conversion and wouldn't be validating the schema "as-is"